huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_attach.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/sdrs/v1/attachreplication"
+	"github.com/chnsz/golangsdk/openstack/sdrs/v1/protectedinstances"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func ResourceReplicationAttach() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceReplicationAttachCreate,
+		ReadContext:   resourceReplicationAttachRead,
+		DeleteContext: resourceReplicationAttachDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceReplicationAttachImportState,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"replication_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"device": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceReplicationAttachCreate(ctx context.Context, d *schema.ResourceData,
+	meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+	// use for WaitForJobSuccess
+	client.Endpoint = client.ResourceBase
+
+	instanceID := d.Get("instance_id").(string)
+	replicationID := d.Get("replication_id").(string)
+	attachOpts := attachreplication.CreateOpts{
+		ReplicationID: replicationID,
+		Device:        d.Get("device").(string),
+	}
+
+	n, err := attachreplication.Create(client, instanceID, attachOpts).ExtractJobResponse()
+	if err != nil {
+		return diag.Errorf("error creating SDRS replication attach: %s", err)
+	}
+
+	createTimeoutSec := int(d.Timeout(schema.TimeoutCreate).Seconds())
+	if err = attachreplication.WaitForJobSuccess(client, createTimeoutSec, n.JobID); err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(formatAttachId(instanceID, replicationID))
+	return resourceReplicationAttachRead(ctx, d, meta)
+}
+
+func resourceReplicationAttachRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	replicationID := d.Get("replication_id").(string)
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
these lines are useless, please check and delete them.

Reply:
OK